README.md

```diff
@@ -81,27 +79,31 @@ gulp.task('lint', function() {
 
 ### Custom reporters
 
-Custom reporter functions can be passed as `csslint.reporter(reporterFunc)`. The reporter function will be called for each linted file and passed the file object as described above.
+Custom reporters can be provided by first adding a valid CSSLint-formatter, such as `csslint-stylish`, then using it:
 
 ```js
 var csslint = require('gulp-csslint');
-var gutil = require('gulp-util');
 
-var customReporter = function(file) {
-  gutil.log(gutil.colors.cyan(file.csslint.errorCount)+' errors in '+gutil.colors.magenta(file.path));
-
-  file.csslint.results.forEach(function(result) {
-    gutil.log(result.error.message+' on line '+result.error.line);
-  });
-};
+csslint.addFormatter(require('csslint-stylish'));
 
 gulp.task('lint', function() {
   gulp.files('lib/*.css')
     .pipe(csslint())
-    .pipe(csslint.reporter(customReporter));
+    .pipe(csslint.reporter('csslint-stylish'))
 });
 ```
 
+You can also provide an object with the following contract to implement your own reporter:
+
+```js
+{
+  id: 'string', // Name passed to csslint.reporter
+  startFormat: function() {}, // Called before parsing any files, should return a string
+  startFormat: function() {}, // Called after parsing all files, should return a string
+  formatResult: function (results, filename, options) {} // Called with a results-object per file linted. Optionally called with a filename, and options passed to csslint.reporter(*reporter*, *options*)
+}
+```
```

Comment:
I can't find it explicitly stated, this is just pulled out from my understanding after looking at all of the existing formatters: https://github.com/CSSLint/csslint/tree/master/src/formatters